apps/web/src/views/Farms/FarmsV3.tsx

```diff
@@ -341,36 +340,38 @@ const Farms: React.FC<React.PropsWithChildren> = ({ children }) => {
   const [numberOfFarmsVisible, setNumberOfFarmsVisible] = useState(NUMBER_OF_FARMS_VISIBLE)
 
   const chosenFarms = useMemo(() => {
-    let chosenFs = []
-    if (isActive) {
-      chosenFs = stakedOnly ? farmsList(stakedOnlyFarms) : farmsList(activeFarms)
-    }
-    if (isInactive) {
-      chosenFs = stakedOnly ? farmsList(stakedInactiveFarms) : farmsList(inactiveFarms)
-    }
-    if (isArchived) {
-      chosenFs = stakedOnly ? farmsList(stakedArchivedFarms) : farmsList(archivedFarms)
-    }
+    let chosenFs: V2StakeValueAndV3Farm[] = []
 
     if (v3FarmOnly || v2FarmOnly || boostedOnly || stableSwapOnly) {
       const filterFarms = chosenFs.filter(
         (farm) =>
           (v3FarmOnly && farm.version === 3) ||
           (v2FarmOnly && farm.version === 2) ||
           (boostedOnly && farm.boosted && farm.version === 3) ||
-          (stableSwapOnly && farm.isStable),
+          (stableSwapOnly && farm.version === 2 && farm.isStable),
       )
 
-      const stakedFilterFarms = chosenFs.filter(
+      const stakedFilterFarms = chosenFs?.filter(
         (farm) =>
+          farm.version === 2 &&
           farm.userData &&
           (new BigNumber(farm.userData.stakedBalance).isGreaterThan(0) ||
-            new BigNumber(farm.userData.proxy?.stakedBalance).isGreaterThan(0)),
+            new BigNumber(farm?.userData.proxy?.stakedBalance ?? 0).isGreaterThan(0)),
       )
 
       chosenFs = stakedOnly ? farmsList(stakedFilterFarms) : farmsList(filterFarms)
     }
 
+    if (isActive) {
```

Comment:
we shouldn't move this part I think. Otherwise the code above is not doing any filtering. Because the `chosenFs` is empty.

Reply:
will endup with problem like this. The farm type filter is not working
<img width="1236" alt="image" src="https://github.com/pancakeswap/pancake-frontend/assets/116779127/c0487d76-9e16-4ad6-969d-f5955cba7ebb">